src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java

```diff
@@ -28,6 +28,18 @@
 
     private final SAMRecord samRecord;
 
+    private transient Integer cachedSoftStart = null;
+    private transient Integer cachedSoftEnd = null;
+    private transient Integer cachedAdaptorBoundary = null;
+    private transient Integer cachedCigarLength = null;
+
+    private void clearCachedValues() {
```

Comment:
I'm a bit worried that repeated calls to clearCachedValues() are going to introduce performance regressions in other places.  I'd sort of lean towards putting the caching in an additional wrapper layer that's only used in haplotype caller, although I understand the concern about that introducing more performance issues.

Reply:
Well, setters are not generally called repeatedly in hotspots, while getters are. So I think it's ok to slightly penalize the setters in order to greatly speed up the getters in critical regions. Putting the caching in a separate layer could harm the performance gains I've seen with this approach, so I'd rather not.